### PR TITLE
deprecation(components/layout): `SkyErrorModalService` is deprecated; use a standard modal with an error component instead

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

File: libs/components/errors/src/lib/modules/error/error-modal.service.ts [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+    @Optional() logService?: SkyLogService
+  ) {
+    this.#_logService = logService;
+  }
   /**
    * Specifies text for the the error message, including title, description, and action label.
+   * @deprecated We recommend using a standard modal with an error component instead.
    */
   public open(config: ErrorModalConfig) {
+    this.#_logService?.deprecated("SkyErrorService's open method", {
```
Comment (johnhwhite) [inline libs/components/errors/src/lib/modules/error/error-modal.service.ts]: This should be `SkyErrorModalService`, right?

**Review by johnhwhite — APPROVED** [review]

File: libs/components/errors/src/lib/modules/error/error-modal.service.ts [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
 /**
  * Opens a modal to display a SKY UX-themed error message.
+ * @deprecated We recommend using a standard modal with an error component instead.
  */
 @Injectable({
   providedIn: 'root',
 })
 export class SkyErrorModalService {
-  constructor(private modal: SkyModalService) {}
+  #_logService: SkyLogService;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/errors/src/lib/modules/error/error-modal.service.ts]: Should be `SkyLogService | undefined`.

File: libs/components/errors/src/lib/modules/error/error-modal.service.ts [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
+    @Optional() logService?: SkyLogService
+  ) {
+    this.#_logService = logService;
+  }
   /**
    * Specifies text for the the error message, including title, description, and action label.
+   * @deprecated We recommend using a standard modal with an error component instead.
    */
   public open(config: ErrorModalConfig) {
+    this.#_logService?.deprecated("SkyErrorModalService's open method", {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/errors/src/lib/modules/error/error-modal.service.ts:29]: Needs a unit test.

File: libs/components/errors/src/lib/modules/error/error-modal.service.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/errors/src/lib/modules/error/error-modal.service.ts]: Fixed

File: libs/components/errors/src/lib/modules/error/error-modal.service.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/errors/src/lib/modules/error/error-modal.service.ts:29]: Added

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**